metadata_service/proxy/neo4j_proxy.py

```diff
@@ -715,9 +719,136 @@ def get_user_detail(self, *, user_id: str) -> Union[UserEntity, None]:
                             team_name=record.get('team_name'),
                             slack_id=record.get('slack_id'),
                             employee_type=record.get('employee_type'),
-                            manager_fullname=manager_record.get('full_name', ''))
+                            manager_fullname=manager_name)
         return result
 
+    @timer_with_counter
+    def get_resources_by_user_relation(self, *, user_id: str, relation: str) -> Dict[str, Any]:
+        """
+        Retrive all the resources bookmarked by user.
+        We start with table resources only, then add dashboard.
+
+        :param user_id: the email of the user
+        :param relation: the relation between the user and the resource
+        :return:
+        """
+
+        # relationship can't be parameterized
+        query_key = 'key: "{user_id}"'.format(user_id=user_id)
+
+        query = textwrap.dedent("""
+        MATCH (user:User {{{key}}})-[:{relation}]->(tbl:Table)
+        RETURN COLLECT(DISTINCT tbl) as table_records
+        """).format(key=query_key,
+                    relation=relation)
+
+        record = self._execute_cypher_query(statement=query,
+                                            param_dict={})
+
+        if not record:
+            raise NotFoundException('User {user_id} does not {relation} '
+                                    'any resources'.format(user_id=user_id,
+                                                           relation=relation))
+        results = []
+        table_records = record.single().get('table_records', [])
+
+        for record in table_records:
+            # todo: decide whether we want to return a list of table entities or just table_uri
+            results.append(self.get_table(table_uri=record['key']))
+        return {'table': results}
+
+    @timer_with_counter
+    def add_resource_relation_by_user(self, *,
+                                      table_uri: str,
+                                      user: str,
+                                      relation: str,
+                                      reverse_relation: str) -> None:
+        """
+        Update table user informations.
+        1. Do a upsert of the user node.
+        2. Do a upsert of the relation/reverse-relation edge.
+
+        :param table_uri:
+        :param user:
+        :param relation:
+        :param reverse_relation:
+        :return:
+        """
+        upsert_user_query = textwrap.dedent("""
+        MERGE (u:User {key: $user_email})
+        on CREATE SET u={email: $user_email, key: $user_email}
+        on MATCH SET u={email: $user_email, key: $user_email}
+        """)
+
+        user_email = 'key: "{user_email}"'.format(user_email=user)
```

Comment:
the reason to do this is because it seems neo4j python client doesn't support relation parameterized.  Hence I need to construct the actual query url before sending over to the actual driver based on different relation ship. I intent to make this function generic as we will have multiple relationship between user and resources(Table, Dashboard, Kafka  Streams etc)